src/Microsoft.AspNet.Razor/Parser/CSharpCodeParser.Statements.cs

```diff
@@ -126,26 +126,35 @@ private void UsingDeclaration()
             // Set block type to directive
             Context.CurrentBlock.Type = BlockType.Directive;
 
-            // Parse a type name
-            Assert(CSharpSymbolType.Identifier);
-            NamespaceOrTypeName();
-            IEnumerable<CSharpSymbol> ws = ReadWhile(IsSpacingToken(includeNewLines: true, includeComments: true));
-            if (At(CSharpSymbolType.Assign))
+            if (At(CSharpSymbolType.Identifier))
             {
-                // Alias
-                Accept(ws);
-                Assert(CSharpSymbolType.Assign);
-                AcceptAndMoveNext();
+                // non-static using
+                NamespaceOrTypeName();
+                IEnumerable<CSharpSymbol> ws = ReadWhile(IsSpacingToken(includeNewLines: true, includeComments: true));
+                if (At(CSharpSymbolType.Assign))
+                {
+                    // Alias
+                    Accept(ws);
+                    Assert(CSharpSymbolType.Assign);
+                    AcceptAndMoveNext();
 
-                AcceptWhile(IsSpacingToken(includeNewLines: true, includeComments: true));
+                    AcceptWhile(IsSpacingToken(includeNewLines: true, includeComments: true));
 
-                // One more namespace or type name
-                NamespaceOrTypeName();
+                    // One more namespace or type name
+                    NamespaceOrTypeName();
+                }
+                else
+                {
+                    PutCurrentBack();
+                    PutBack(ws);
+                }
             }
-            else
+            else if (At(CSharpKeyword.Static))
             {
-                PutCurrentBack();
-                PutBack(ws);
+                // static using
+                AcceptAndMoveNext();
+                AcceptWhile(IsSpacingToken(includeNewLines: false, includeComments: true));
```

Comment:
Why must the identifier be on the same line as `@using static`? In a regular .cs file, the following is fine:
``` c#
using
    static
    System.Console;
```